src/util.ts

```diff
@@ -297,6 +297,10 @@ function escapePathAccess(string: string) {
   return string.replace(/(\[|\]|\.|'|")/g, '\\$1');
 }
 
+export function escapeSingleQuotes(value: string) {
```

Comment:
It seems like a code smell that we need to add a new `escapeSingleQuotes` utility method.  

Shouldn't we already have another escape utility in other places? (Also, for more than just "single quotes"?)

Reply:
I tried other utility functions to re-escape the field name but could not find a good one.  Maybe I will try to find again and ping you if I can't find.

Reply:
There may be some similar logic in Vega (utils).

Reply:
Ended up with creating more generic function, `accessWithDatumToUnescapedPath`, to solve this case.
`accessWithDatumToUnescapedPath` is taking un-escaped string to build the datum-access expression.